packages/core/src/compiler/Style.ts

```diff
@@ -567,22 +567,31 @@ const mergeMapping = (
     throw Error("var has no binding form?");
   }
   const [, bindingForm] = res;
-
   switch (bindingForm.tag) {
-    case "SubVar": {
-      // G || (x : |T) |-> G
-      return varEnv;
-    }
-    case "StyVar": {
-      // G || (y : |T) |-> G[y : T] (shadowing any existing Sub vars)
+    case "StyVar":
       return {
         ...varEnv,
         vars: varEnv.vars.set(
           bindingForm.contents.value,
           toSubstanceType(styType)
         ),
+        varIDs: [
+          dummyIdentifier(bindingForm.contents.value, "Style"),
+          ...varEnv.varIDs,
+        ],
+      };
+    case "SubVar":
+      return {
+        ...varEnv,
+        vars: varEnv.vars.set(
+          bindingForm.contents.value,
+          toSubstanceType(styType)
+        ),
```

Comment:
This code looks duplicated from the above code for `StyVar`; does it make sense to pull it to above the `switch`?

Reply:
Done!